overlord/snapstate/snapstate.go

```diff
@@ -55,8 +55,12 @@ func needsMaybeCore(typ snap.Type) int {
 }
 
 func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int) (*state.TaskSet, error) {
-	if snapsup.Flags.Classic && !release.OnClassic {
-		return nil, fmt.Errorf("classic confinement is only supported on classic systems")
+	if snapsup.Flags.Classic {
+		if !release.OnClassic {
+			return nil, fmt.Errorf("classic confinement is only supported on classic systems")
+		} else if !release.ReleaseInfo.SupportsClassicSnaps() {
+			return nil, fmt.Errorf("classic confinement is not supported on your distribution")
```

Comment:
We need to fix it. This should never be an optional thing as its a fundamental thing of the snap experience.